mxnet/training/docker/1.9/py3/cu112/Dockerfile.gpu

```diff
@@ -42,7 +42,8 @@ ENV PYTHONDONTWRITEBYTECODE=1 \
     DGLBACKEND=mxnet \
     CUDNN_VERSION=8.1.0.77-1+cuda11.2 \
     NCCL_VERSION=2.8.3-1+cuda11.2 \
-    DEBIAN_FRONTEND=noninteractive
+    DEBIAN_FRONTEND=noninteractive \
+    DLC_CONTAINER_TYPE=training
```

Comment:
Why do we have training images in this PR?

Reply:
I brought in the commits from this PR to resolve some of the test failures.
https://github.com/aws/deep-learning-containers/pull/2458

If we merge this PR, we can just close the other and knock out both fixes in one go.

If this isn't something you'd like to do, feel free to revert and we can look at merging the other PR as well.

Reply:
Can we revert all the training Dockerfile changes and handle them in a different PR? I say this because we have not run any tests on the training till now.